### DISCUSSION TITLE
About all gems reachable

### DISCUSSION THREAD

**Discussion by tcrl** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#22] [Q&A] [answered]
The readme file states that only Walls need to be considered as to whether a Gem is reachable by the Player using the flood fill algorithm, but that we are "free to add to this algorithm to provide more robust checking of whether a gem is reachable". I am uncertain as to how strict we have to implement this check.

For example, is the Gem in the following case reachable? There are no walls blocking the Player and the flood fill algorithm fills the Gem-containing cell fine as it moves step by step; but due to the lack of Stop Cells, the Player can't actually reach the cell with the Gem.
P..
.G.
...

Edit: I realised that the obfuscated jar also marks the above situation as reachable, but I would still like to clarify whether/what kind of more robust checking may be required on top of the flood fill algorithm.

Thanks!

**Comment by Derppening** [comment, marked as answer]
For your example, your implementation is free to accept or reject the gameboard (in fact this issue was raised in #4).

As you mentioned, the flood-fill algorithm will deem this board as not violating the "all gems reachable" constraint, whereas a stricter implementation (e.g. the one which also checks for stop cells) will deem this board as violating the constraint.

However, I can guarantee that no "all gems reachable" test cases will fail if a stricter implementation is written. If there is an issue with the final grading regarding this implementation, feel free to request an appeal after final grading.

**Reply by tcrl** [reply]
Just to double-check, are there any hidden cases where the standard flood fill algorithm implementation would not suffice? Thanks for your swift reply!

**Reply by Derppening** [reply]
No. All "all gems reachable" test cases will pass with a naive flood-fill algorithm.
